AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLConnectionScenario3Tests+Subscribe.swift

```diff
@@ -0,0 +1,232 @@
+        }
+
+        let uuid2 = UUID().uuidString
+        guard createPost(id: uuid2, title: title) != nil else {
+            XCTFail("Failed to create post")
+            return
+        }
+
+        wait(for: [progressInvoked], timeout: TestCommonConstants.networkTimeout)
+        operation.cancel()
+        wait(for: [disconnectedInvoked, completedInvoked], timeout: TestCommonConstants.networkTimeout)
+        XCTAssertTrue(operation.isFinished)
+    }
+
+    func testOnUpdatePostSubscriptionWithModel() {
+        let connectedInvoked = expectation(description: "Connection established")
+        let disconnectedInvoked = expectation(description: "Connection disconnected")
+        let completedInvoked = expectation(description: "Completed invoked")
+        let progressInvoked = expectation(description: "progress invoked")
+
+        let operation = Amplify.API.subscribe(
+            request: .subscription(of: Post3.self, type: .onUpdate),
+            valueListener: { event in
+                switch event {
+                case .connection(let state):
+                    switch state {
+                    case .connecting:
+                        break
+                    case .connected:
+                        connectedInvoked.fulfill()
+                    case .disconnected:
+                        disconnectedInvoked.fulfill()
+                    }
+                case .data:
+                    progressInvoked.fulfill()
+                }
+        },
+            completionListener: { event in
+                switch event {
+                case .failure(let error):
+                    print("Unexpected .failed event: \(error)")
+                case .success:
+                    completedInvoked.fulfill()
+                }
+        })
+        XCTAssertNotNil(operation)
+        wait(for: [connectedInvoked], timeout: TestCommonConstants.networkTimeout)
+        let uuid = UUID().uuidString
+        let testMethodName = String("\(#function)".dropLast(2))
+        let title = testMethodName + "Title"
+
+        guard createPost(id: uuid, title: title) != nil else {
+            XCTFail("Failed to create post")
+            return
+        }
+
+        guard updatePost(id: uuid, title: title) != nil else {
+            XCTFail("Failed to update post")
+            return
+        }
+
+        wait(for: [progressInvoked], timeout: TestCommonConstants.networkTimeout)
+        operation.cancel()
+        wait(for: [disconnectedInvoked, completedInvoked], timeout: TestCommonConstants.networkTimeout)
+        XCTAssertTrue(operation.isFinished)
+    }
+
+    func testOnDeletePostSubscriptionWithModel() {
+        let connectedInvoked = expectation(description: "Connection established")
+        let disconnectedInvoked = expectation(description: "Connection disconnected")
+        let completedInvoked = expectation(description: "Completed invoked")
+        let progressInvoked = expectation(description: "progress invoked")
```

Comment:
i'm guessing progress invoked is only invoked once here because we are doing 1 create and 1 delete?

Reply:
i believe it is because the subscription is `onDelete` mutations only so only 1 delete will corespond to the subscription event